src/Components/Assets/AssetManage.tsx

```diff
@@ -0,0 +1,273 @@
+  const [offset, setOffset] = useState(0);
+  const [totalCount, setTotalCount] = useState(0);
+  const [transactions, setTransactions] = useState<AssetTransaction[]>([]);
+  const [transactionDetails, setTransactionDetails] = useState<
+    ReactElement | ReactElement[]
+  >();
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const dispatch = useDispatch();
+  const limit = 14;
+
+  const fetchData = useCallback(
+    async (status: statusType) => {
+      setIsLoading(true);
+      const [assetData, transactionsData]: any = await Promise.all([
+        dispatch(getAsset(assetId)),
+        dispatch(listAssetTransaction({ asset: assetId, limit, offset })),
+      ]);
+      if (!status.aborted) {
+        setIsLoading(false);
+        if (!assetData.data)
+          Notification.Error({
+            msg: "Something went wrong..!",
+          });
+        else {
+          setAsset(assetData.data);
+          setTransactions(transactionsData.data.results);
+          setTotalCount(transactionsData.data.count);
+        }
+      }
+    },
+    [dispatch, assetId, offset]
+  );
+
+  useAbortableEffect(
+    (status: statusType) => {
+      fetchData(status);
+    },
+    [dispatch, fetchData]
+  );
+
+  const handlePagination = (page: number, limit: number) => {
+    const offset = (page - 1) * limit;
+    setCurrentPage(page);
+    setOffset(offset);
+  };
+
+  const PrintPreview = () => (
+    <div className="">
+      <div className="my-4 flex justify-end ">
+        <button
+          onClick={(_) => window.print()}
+          className="btn btn-primary mr-2"
+        >
+          <i className="fas fa-print mr-2"></i> Print QR Code
+        </button>
+        <button
+          onClick={(_) => setIsPrintMode(false)}
+          className="btn btn-default"
+        >
+          <i className="fas fa-times mr-2"></i> Close
+        </button>
+      </div>
+      <h2 className="text-center">Print Preview</h2>
+      <div id="section-to-print" className="print flex justify-center">
+        <QRCode size={200} value={asset?.id || ""} />
+      </div>
+    </div>
+  );
+
+  const working_status = (is_working: boolean | undefined) => {
+    if (is_working)
+      return (
+        <span className="bg-green-500 text-white text-sm px-2 py-1 uppercase rounded-full">
+          Working
+        </span>
+      );
+    return (
+      <span className="bg-red-500 text-white text-sm px-2 py-1 uppercase rounded-full">
+        Not Working
+      </span>
+    );
+  };
+
+  const status = (
+    asset_status: "ACTIVE" | "TRANSFER_IN_PROGRESS" | undefined
+  ) => {
+    if (asset_status === "ACTIVE") {
+      return (
+        <span className="bg-green-500 text-white text-sm px-2 py-1 uppercase rounded-full">
+          ACTIVE
+        </span>
+      );
+    }
+    return (
+      <span className="bg-yellow-500 text-white text-sm px-2 py-1 uppercase rounded-full">
+        TRANSFER IN PROGRESS
+      </span>
+    );
```

Comment:
Is there any relation between `ACTIVE` and `WORKING`. Is it possible that assets is not active but is working or vice versa ?

Reply:
Yeah I guess active is the current state of the asset but working is more of a general thing.